### ISSUE TITLE
Could not pull content from connection due to error. (external connection)

### ISSUE THREAD

**Issue by msaidizi** [10up/distributor#487]
Hi
I have this issue and I have tried to read all the info available here and although this issue has been raised multiple times it seems the solutions I have tried are not working. 
1. I have a multisite setup with subdomains ([bettinghits.com](https://bettinghits.com))and a single external site. ([sportpesagames.com](https://sportpesagames.com)) on the same server.
2. _I am trying to pull content from the single site into a subdomain._
3. I have installed distributor on both the multisite and single site. The multisite set up is working beautifully but the external one is not.
4. I have installed the password plugin and I set up a password and name on both sides (different from my wp login and password) 
5. I am using the theme [gist](https://wordpress.org/themes/gist/) on both sites and I have no custom post types.
6. I have edited my htaacess to reflect the remote access option. (I was not getting the error but I thought I would try that since it was a solution added here.
7. I am the only admin on both sites and yes I have made sure I am the super admin in multisite and admin in the other site.
8. This is the error when I try to connect 
Limited connection established. Authentication failed.
Push distribution unavailable.
Pull distribution limited to basic content, i.e. title and content body.
9. Version 1.5.0
10. I also would like to disable the standard conical element, Is there a novice tutorial because i have no idea where this is **removing Distributor's default front end canonical URL filtering (look for 'get_canonical_url' and 'wpseo_canonical').**
Thanks a bunch

**Comment by shockthealien** [comment]
I'm having a similar issue tho my setup is slightly different. (I'm using Themify Post Type Builder to create custom post types). Do you get this error too on the connection? All my credentials are correct and still get this all the time
<img width="464" alt="Screenshot 2019-11-11 at 10 09 26" src="https://user-images.githubusercontent.com/2083273/68579366-d0d73d00-046b-11ea-9112-ed148a796c80.png">

**Comment by msaidizi** [comment]
yes am getting this same error

> I'm having a similar issue tho my setup is slightly different. (I'm using Themify Post Type Builder to create custom post types). Do you get this error too on the connection? All my credentials are correct and still get this all the time
> <img alt="Screenshot 2019-11-11 at 10 09 26" width="464" src="https://user-images.githubusercontent.com/2083273/68579366-d0d73d00-046b-11ea-9112-ed148a796c80.png">

**Comment by jeffpaul** [comment]
@msaidizi and @shockthealien please confirm that you're following the precise steps to [set up External Connections using Application Passwords](https://github.com/10up/distributor#setup-external-connections-using-application-passwords) as these should be the exact steps necessary to create fully functioning non-multisite connections for Distributor.

**Comment by shockthealien** [comment]
@jeffpaul yes Thats the instructions i followed to set it up

**Comment by msaidizi** [comment]
me too

**Comment by msaidizi** [comment]
> @jeffpaul yes Thats the instructions i followed to set it up

I deleted everything and refollowed those instructions step by step and I got it to work....

**Comment by shockthealien** [comment]
@jeffpaul I deleted everything too and still not working. Does it have to use the Application Passwords plugin? it keeps giving me this error even tho I have performed the fix suggested
<img width="858" alt="Screenshot 2019-11-13 at 11 01 13" src="https://user-images.githubusercontent.com/2083273/68758204-432f5500-0605-11ea-8560-a2c865bc3a97.png">

**Comment by shockthealien** [comment]
@jeffpaul any advice on this? need to decide on which route to take to build the group of websites and I have the directors breathing down my neck to get moving

**Comment by shockthealien** [comment]
BTW i have scrapped Themify PTB and im just using normal posts now with a couple custom meta fields

**Comment by jeffpaul** [comment]
@shockthealien we strongly recommend using the Application Passwords plugin on the site you are connecting to in order to create a unique password for Distributor's connections.  Alternatively, you can create an application with the WordPress.com applications manager as the authentication method.

**Comment by shockthealien** [comment]
@jeffpaul thanks. I'm not against using the plugin, I just don't seem to be able to get it to work even performing the troubleshooting they suggest to fix it, so I don't really know how to fix that.

I looked into the WP.com application and i don't know what half of the things it asks for are so I doubt that is the way forward

**Comment by jeffpaul** [comment]
@shockthealien we're working to add more and better error messaging in #269, ideally targeted for our next Distributor release, that would help me further troubleshoot this for you.  Can you confirm that you're using the Application Passwords user password and not Account Management user password?  Can you also confirm that you're adding `wp-json` to the end of the External Connection URL?

**Comment by shockthealien** [comment]
@jeffpaul thanks for the reply. Yes, I am using the password created with the Application Passwords plugin and not the WP user password. And yes I am adding the wp-json to the URL.

<img width="1120" alt="Screenshot 2019-11-21 at 15 28 56" src="https://user-images.githubusercontent.com/2083273/69351767-b8d59980-0c73-11ea-9032-fc18e0dfd7c3.png">

I'm sure this all stems from the AP plugin not working properly since it is still showing the error i posted about earlier. I have reinstalled it and performed their suggested fix but unfortunately,  it is still showing the same error

**Comment by shockthealien** [comment]
For more context, the setup I'm working with is one central site (mainsite as per your instructions?) for all news articles for the group (http://checkmark.ateb-group.co.uk) and four other websites for different parts of the business and I need to push only the relevant news to these sites. Currently I have one of them I'm trying to get this working with (http://home.ateb-group.co.uk)(remotesite?).

I have distributor and AP on both sites installed. I have edited the .htaccess file on both as per AP instructions too

**Comment by jeffpaul** [comment]
@shockthealien thanks for giving that a try, sorry that this hasn't gone smoothly for you.  Can you confirm the plugins and theme you have installed and active?  That won't likely show an immediate cause, but it may help troubleshooting in the future if we know there's a plugin/theme compatibility issue with Distributor.

**Comment by jeffpaul** [comment]
@shockthealien are the checkmark and home subdomain sites set up within a multisite instance or are these separate WordPress instances but mapped to the same domain?

**Comment by shockthealien** [comment]
no worries. Thanks for trying to help.

I am Using Themify Builder pro on all of the sites. The checkmark one is using themify ultra theme, heavily customised. The group one is using Themify Basic theme and pretty much built from scratch from there.

They are separate independent WP instances on each subdomain. Also they are not likely the final domains. On release they will likely each have their own proper full domain

**Comment by shockthealien** [comment]
@jeffpaul plugin wise, theres quite a few. Ive Screenshotted the lists. Hope that helps
<img width="1607" alt="Screenshot 2019-11-21 at 16 01 33" src="https://user-images.githubusercontent.com/2083273/69355070-17514680-0c79-11ea-8c50-102d722c600f.png">
<img width="1607" alt="Screenshot 2019-11-21 at 16 01 14" src="https://user-images.githubusercontent.com/2083273/69355071-17514680-0c79-11ea-8e06-daf5c3539709.png">

**Comment by shockthealien** [comment]
@jeffpaul just as an experiment I tried setting up one of the other sites i hadnt touched yet, fresh WP installation. Just AP and Distributor installed and getting the same issue. So i dont think the plugins have anything to do with this.
<img width="1156" alt="Screenshot 2019-11-22 at 10 00 50" src="https://user-images.githubusercontent.com/2083273/69416771-40241b00-0d0f-11ea-8b6f-9b1687991045.png">

**Comment by shockthealien** [comment]
@jeffpaul any updates on this issue? I am really stuck with this and not sure how to move it forward. Is the wordpress application an alternative way of doing the same? if so, any chance of some help setting that up?

**Comment by jeffpaul** [comment]
@shockthealien no, I haven't been able to setup a replica environment yet to try and reproduce your issues to try and find a solution.  I don't have documentation on the WP.com app set up process, though if you give that a try and take some notes I'd gladly turn those into step-by-step instructions for our docs.

**Comment by shockthealien** [comment]
@jeffpaul Thanks. I could grant you access to my environment if that helps?

**Comment by Youdaman** [comment]
Same problem with two fresh vanilla installs of WordPress on local server, i.e. two domains both with 127.0.0.1 -- limited/yellow connection status as per conversation above and then "Could not pull content from connection due to error." when clicking on the pull menu item (note: using user/pass method as wordpress.com method does not work for locally hosted sites).

**Comment by jeffpaul** [comment]
Thanks for the input @Youdaman, are you seeing any console errors, potential conflicting plugins or configurations that could help further triage this issue?

**Comment by Youdaman** [comment]
None unfortunately. Steps to repro were just unzip the latest wordpress twice, do two fresh installs, install only distributor on both and try to use it. I'm on Windows 10 if that matters with Apache 2.4.20 and PHP 7.2.9

**Comment by Thakurpritika** [comment]
Hi I'm getting the same error
![image](https://user-images.githubusercontent.com/130210999/230709456-822e2731-af08-47c1-91f9-5d39dcf893e8.png)

**Comment by Thakurpritika** [comment]
![image](https://user-images.githubusercontent.com/130210999/230709541-bf0053f0-ca24-4fd0-8362-a0250ed3d233.png)
